validator/sawtooth_validator/networking/dispatch.py

```diff
@@ -174,48 +174,40 @@ def _process(self, message_id):
             del self._message_information[message_id]
             return
 
-        future = handler_manager.execute(connection_id, message.content)
-
         timer_tag = type(handler_manager.handler).__name__
         timer_ctx = self._get_dispatch_timer(timer_tag).time()
 
-        def do_next(future):
+        def do_next(result):
             timer_ctx.stop()
             try:
-                self._determine_next(message_id, future)
+                self._determine_next(message_id, result)
             except Exception:  # pylint: disable=broad-except
                 LOGGER.exception(
                     "Unhandled exception while determining next")
 
-        future.add_done_callback(do_next)
-
-    def _determine_next(self, message_id, future):
-        try:
-            res = future.result(timeout=self._timeout)
-        except TimeoutError:
-            LOGGER.exception("Dispatcher timeout waiting on handler result.")
-            raise
+        handler_manager.execute(connection_id, message.content, do_next)
 
-        if res is None:
+    def _determine_next(self, message_id, result):
+        if result is None:
```

Comment:
the rename of result should be a separate commit.

Reply:
What do you want instead? `_determine_next(self, message_id, res)`?

Reply:
I like the rename, I just think the rename and the move to a parameter could be two separate commits.  It was hard to see the actual change, due to the renaming clutter.

Reply:
I pushed the renaming into the refactor commit.